graphql/http_handler.go

```diff
@@ -23,7 +30,34 @@ func Handler(apiURL string) func(w http.ResponseWriter, r *http.Request) {
 	// Send OTEL traces for each request.
 	// Only create spans for resolved fields.
 	srv.Use(otelgqlgen.Middleware(
-		otelgqlgen.WithCreateSpanFromFields(func(fieldCtx *graphql.FieldContext) bool { return fieldCtx.IsResolver }),
+		otelgqlgen.WithCreateSpanFromFields(func(fieldCtx *graphql.FieldContext) bool {
+			return fieldCtx.IsMethod
+		}),
+		otelgqlgen.WithRequestVariablesAttributesBuilder(
+			otelgqlgen.RequestVariablesBuilderFunc(func(requestVariables map[string]interface{}) []attribute.KeyValue {
+				// Deep clone the request variables to avoid modifying the original map.
```

Comment:
Where would the map get modified?

Reply:
So RedactedFieldsInMap does an in place update of the `requestVariables`. When writing this I realized that the otelgqlgen library passes a reference to the actual variables the graphql library uses to execute the underlying resolvers. If we don't create a deep clone of the object we risk updating the request variables the resolvers receive and end up returning broken or invalid data.